geco/generator.py

```diff
@@ -17,3 +18,41 @@ def __next__(self):
 
     def seed(self, seed: int):
         self.rng.seed(seed)
+
+
+@py_random_state(-1)
+def common_substructure_generator(
+    instance_generation_function,
+    instance_params_generation_function,
+    base_params,
+    new_params,
+    expand_params_function,
+    seed=0,
+):
+    """
+    Generates instances that have common substructure
+
+    Parameters
+    ----------
+    instance_generation_function:
+        base function that defines MIP
+    instance_params_generation_function:
```

Comment:
`instance_parmas_generation_function` and `base_params` are single use. I think we can merge them for readability and easier testing into one parameter called `backbone`.

Reply:
Please check if the current implementation is the one you had in mind